From 596cf4d04580b191d2f4f6082000534bdab13791 Mon Sep 17 00:00:00 2001
From: Vishnu Banavath <vishnu.banavath@arm.com>
Date: Sat, 11 Dec 2021 13:23:55 +0000
Subject: [PATCH 17/24] efi_loader: corstone1000: remove guid check from
 corstone1000 config option

Use generic fmp guid and no separte check is required for
CORSTONE1000 target.

Signed-off-by: Vishnu Banavath <vishnu.banavath@arm.com>
Signed-off-by: Rui Miguel Silva <rui.silva@linaro.org>
---
 lib/efi_loader/efi_capsule.c | 16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

diff --git a/lib/efi_loader/efi_capsule.c b/lib/efi_loader/efi_capsule.c
index e08e97cf3fb7..891143c33909 100644
--- a/lib/efi_loader/efi_capsule.c
+++ b/lib/efi_loader/efi_capsule.c
@@ -657,12 +657,6 @@ efi_status_t __efi_runtime EFIAPI efi_update_capsule(
 			  i, &capsule->capsule_guid);
 
 #if CONFIG_IS_ENABLED(TARGET_CORSTONE1000)
-		if (guidcmp(&corstone1000_capsule_guid, &capsule->capsule_guid)) {
-			ret = EFI_INVALID_PARAMETER;
-			log_err("Corstone1000: Invalid capsule GUID\n");
-			goto out;
-		}
-
 		if (efi_size_in_pages(capsule->capsule_image_size) >
 		    CORSTONE1000_CAPSULE_BUFFER_SIZE) {
 			log_err("Corstone1000: Capsule data size exceeds the shared buffer size\n");
@@ -688,15 +682,7 @@ efi_status_t __efi_runtime EFIAPI efi_update_capsule(
 		goto out;
 #endif
 
-		if (!guidcmp(&capsule->capsule_guid,
-			     &efi_guid_firmware_management_capsule_id)) {
-			ret  = efi_capsule_update_firmware(capsule);
-		} else {
-			log_err("Unsupported capsule type: %pUs\n",
-				&capsule->capsule_guid);
-			ret = EFI_UNSUPPORTED;
-		}
-
+		ret  = efi_capsule_update_firmware(capsule);
 		if (ret != EFI_SUCCESS)
 			goto out;
 	}
-- 
2.37.1

